src/core/server/elasticsearch/cluster_client.ts

```diff
@@ -82,6 +86,17 @@ async function callAPI(
   }
 }
 
+/**
+ * Fake request object created manually by Kibana plugins.
+ * @deprecated
```

Comment:
Now I'm not sure we should deprecate it, as it's already a common pattern to perform a request on behalf of a user.

Reply:
Until we have an alternative strategy, I think we should leave this undeprecated.